reconcile/openshift_base.py

```diff
@@ -135,94 +149,109 @@ def init_specs_to_fetch(
                         f"{cluster}/{namespace} (valid kinds: {managed_types})"
                     )
 
-            for kind, names in resource_names.items():
-                c_spec = StateSpec(
-                    "current",
-                    oc,
-                    cluster,
-                    namespace,
-                    kind,
-                    resource_type_override=resource_type_overrides.get(kind),
-                    resource_names=names,
-                )
-                state_specs.append(c_spec)
-                managed_types.remove(kind)
-
-            # Produce "empty" StateSpec's for any resource type that
-            # doesn't have an explicit managedResourceName listed in
-            # the namespace
-            state_specs.extend(
-                StateSpec(
-                    "current",
-                    oc,
-                    cluster,
-                    namespace,
-                    t,
-                    resource_type_override=resource_type_overrides.get(t),
-                    resource_names=None,
+            # Initialized current state specs
+            for kind in managed_types:
+                managed_names = resource_names.get(kind)
+                kind = resource_type_overrides.get(kind, kind)
```

Comment:
I'd rather not override the loop variable, TBH.

Reply:
[7641249](https://github.com/app-sre/qontract-reconcile/pull/2356/commits/7641249f9c0b47138353c5239d33bc819a442b1e)